File: monNSRothC_1.0_matlab/rhofun.m
function out=rhofun(temp,rain,evap,s_cover,clay)

n=length(temp);

% the rate modifying factor for temperature
a = 47.91./ ( 1+ exp( 106.06 ./ (temp+18.27) ) );

% the rate modifying factor for moisture
maxTSMD = -(20 + 1.3*clay-0.01*clay^2);
vec=rain-0.75*evap;
accTSMD=zeros(n,1);
accTSMDf=zeros(n,1);
i=1;
while vec(i)>0
    i=i+1;
end
for j=i:n
    accTSMD(j)= min( max( accTSMD(j-1)+vec(j) , maxTSMD ), 0);
end
b=ones(n,1);
ind= accTSMD<=0.444*maxTSMD;
b(ind) = 0.2+ (1-0.2)* (maxTSMD-accTSMD(ind))./(maxTSMD-0.444*maxTSMD);

c=ones(n,1);
c(s_cover==1)=0.6;

out = a.*b.*c;
end

File: NSRothC_2.0 R/rhofun.R
rhofun <- function(temp, rain, pet, cover, clay, d, meantemp) {
  n <- length(temp)
  
  # The rate modifying factor for temperature
  a <- 47.91 / (1 + exp(106.06 / (temp - meantemp + 106.06 / log(46.91))))
  
  # The rate modifying factor for moisture
  Mcover <- -(20 + 1.3 * clay - 0.01 * clay^2) * d / 23
  Mbare <- Mcover / 1.8
  vec <- rain - pet
  accTSMD <- rep(0, n)
  i <- 1
  while (vec[i] > 0) {
    i <- i + 1
  }
  for (j in i:n) {
    if (cover[j] == 0) {
      accTSMD[j] <- min(max(accTSMD[j - 1] + vec[j], Mcover), 0)
    } else {
      accTSMD[j] <- min(max(accTSMD[j - 1] + vec[j], Mbare), 0)
    }
  }
  
  b <- rep(1, n)
  indc <- cover == 0 & accTSMD <= 0.444 * Mcover
  indb <- cover == 1 & accTSMD <= Mbare
  b[indc] <- 0.2 + (1 - 0.2) * (Mcover - accTSMD[indc]) / (Mcover - 0.444 * Mcover)
  b[indb] <- 0.2 + (1 - 0.2) * (Mbare - accTSMD[indb]) / (Mbare - 0.444 * Mbare)
  
  c <- rep(1, n)
  c[cover == 1] <- 0.6
  
  return(list(a = a, b = b, c = c, accTSMD = accTSMD))
}
